src/runtime/virtcontainers/qemu.go

```diff
@@ -812,6 +813,59 @@ func (q *qemu) setupVirtioMem(ctx context.Context) error {
 	return err
 }
 
+// setupEarlyQmpConnection creates a listener socket to be passed to QEMU
+// as a QMP listening endpoint. An initial connection is established, to
+// be used as the QMP client socket. This allows to detect an early failure
+// of QEMU instead of looping on connect until some timeout expires.
+func (q *qemu) setupEarlyQmpConnection() (net.Conn, error) {
+	monitorSockPath := q.qemuConfig.QMPSockets[0].Name
```

Comment:
Why is it `QMPSockets[0]`? Is there any situation where we expect multiple QMP sockets?

Reply:
The array was added in `govmm` before it got imported into the main `kata-containers` repo (see https://github.com/kata-containers/govmm/commit/6fe338d604357b10944220d0cac39f5712ba3f5e). I agree with you that this doesn't seem to be needed by the current code base. Maybe we can get rid of the complexity in a subsequent PR.